packages/angular_devkit/benchmark/src/main_spec.ts

```diff
@@ -142,4 +145,51 @@ describe('benchmark binary', () => {
     stdout.lines.forEach(line => expect(line).toMatch(/^\[abc\]/));
     expect(res).toEqual(0);
   });
+
+  it('uses watch-script and watch-matcher', async () => {
+    const args = [
+      '--watch-matcher',
+      'Complete',
+      '--watch-script',
+      watchTriggerScript,
+      '--',
+      'node',
+      benchmarkWatchScript,
+    ];
+    const res = await main({ args, stdout, stderr });
+    expect(stdout.lines).toContain('[benchmark] Process Stats\n');
+    expect(res).toEqual(0);
+  }, 30000);
+
+  it('should not fail with exit code', async () => {
+    const args = [
+      '--watch-matcher',
+      'Complete',
+      '--watch-script',
+      watchTriggerScript,
+      '--',
+      'node',
+      exitCodeOneScript,
+    ];
+    const res = await main({ args, stdout, stderr });
+    expect(stderr.lines).toContain('[benchmark] Maximum number of retries (5) for command was exceeded.\n');
+    expect(res).toEqual(1);
+  });
+
+  it('should error when watch-timeout is exceeded', async () => {
```

Comment:
Maybe we should treat timeouts as consuming an individual retry... I guess that depends on how flaky the thing is. On webpack we often have trouble with writes being detected and that hits the timeout, failing the whole thing.

We don't have to do that now though. We can see if this model is appropriate while benchmarking stuff and if it needs to be more robust, we tweak it.

Reply:
This is the current implementation. A failed iteration will consume a retry.

Reply:
This is the current implementation. A failed iteration will consume a retry.